src/api/nudgeStale.ts

```diff
@@ -0,0 +1,137 @@
+   * it would have done. Useful for testing.
+   */
+  dryRun?: boolean
+
+  /**
+   * Optional logging function that will be called with progress messages
+   *
+   * @param message - The message to log
+   */
+  logger?: (message: string) => void
+}
+
+/**
+ * Finds auto-translation PRs that are stale, eg the maintainers of the locales have
+ * not responded to pull request review requests for X days, and nudges them with a
+ * comment for the maintainers indicating that it's been a while without activity.
+ * Also adds a label to the PR to indicate that a nudge has happened, preventing it
+ * from doing the same operation again in the future.
+ *
+ * @param options - Options for the nudge operation
+ * @returns A promise that resolves when operation is complete
+ * @internal
+ */
+export async function nudgeStale(options: NudgeStaleOptions): Promise<void> {
+  const {dryRun, logger = noop} = options
+
+  if (dryRun) {
+    logger('Running in dry-run mode, no actions will be executed')
+  }
+
+  logger('Finding pending auto-translated PRs')
+  const [locales, pendingPRs] = await Promise.all([
+    getLocaleRegistry(),
+    findPendingStaleAutoTranslatedPRs(),
+  ])
+
+  // Filter down to the PRs that are _stale_, eg beyond our threshold
+  const now = new Date()
+  const daysSinceDate = (date: string) =>
+    Math.floor((now.getTime() - new Date(date).getTime()) / ONE_DAY_MS)
+
+  const stale = pendingPRs.filter((pr) => daysSinceDate(pr.createdAt) > STALE_NUDGE_THRESHOLD_DAYS)
+  logger(`Found ${pendingPRs.length} pending PRs`)
+  logger(`Found ${stale.length} stale PRs`)
+  if (stale.length === 0) {
+    return
+  }
+
+  // For each PR, comment on it
+  for (const pr of stale) {
+    logger(`Processing PR #${pr.number}`)
+    const localeId = pr.headRefName.slice(AUTO_TRANSLATE_BRANCH_PREFIX.length + 1)
+    const locale = locales.find((candidate) => candidate.id === localeId)
+    if (!locale) {
+      logger(`No locale found for ${localeId}, skipping`)
+      continue
+    }
+
+    logger(
+      `Commenting on PR #${pr.number} for locale ${localeId} (${locale.englishName || locale.name})`,
+    )
+    await commentOnStalePR(pr, locale, options)
+  }
+}
+
+async function commentOnStalePR(
+  pr: GitHubPRList[number],
+  locale: Locale,
+  {dryRun, logger = noop}: NudgeStaleOptions,
+) {
+  const rootPath = await getRootPath()
+  const execOptions = {cwd: rootPath, env: {...process.env, CLICOLOR: '0'}}
+  const comment = getCommentBody(locale)
+  const addLabel = PR_LABEL_NUDGED
+
+  if (dryRun) {
+    logger(`Would have commented on PR #${pr.number} with:\n\`\`\`\n${comment}\n\`\`\``)
+    logger(`Would have added "${addLabel}" label on PR #${pr.number}`)
+    return
+  }
+
+  await execFile('gh', ['pr', 'comment', `${pr.number}`, '--body', comment], execOptions)
+  await execFile('gh', ['pr', 'edit', `${pr.number}`, '--add-label', addLabel], execOptions)
```

Comment:
If we swap these operations at least we don’t risk pinging someone daily if there is a problem with the edit command

Reply:
Great point!